pkg/plugins/connection/ssh.go

```diff
@@ -40,13 +42,39 @@ func (t *SSHOpen) Metadata() *gornir.TaskMetadata {
 	return t.Meta
 }
 
+func configPubKey(host *gornir.Host) (*ssh.Signer, error) {
+	// var hostKey ssh.PublicKey
+	sshPrivKeyFname, ok := host.Data["ssh_private_key_file"].(string)
```

Comment:
not sure if there is an easy way of reading this from `~/.ssh/config` instead. I think it makes more sense to try to read from well-known locations or to read from the ssh config to avoid having to add the data in the inventory.

Reply:
Are you okay with including something like this?  https://github.com/kevinburke/ssh_config. Do we want to allow the user to override certain values via the inventory?

Reply:
>  https://github.com/kevinburke/ssh_config

Yeah, certainly better than doing it ourselves :P

Something that just occurred to me, it'd be nice to add a concurrency-safe "cache" to gornir so we can store the parsed ssh config object to avoid parsing it each time. Not sure if you want to give it a stab as well

> Do we want to allow the user to override certain values via the inventory?

Unless you really need it I'd say we don't even bother discussing it (less tests, less docs and less stuff to maintain in general :P)

Reply:
Ping me on the NTC slack next week to chat about the cache; sounds like that is worthy of a separate PR.  I agree on the override after sleeping on it - that's what the config file is for in the first place.  :)  I could memoize the config and signer functions, but that's a specific solution to a more general problem.